### PR TITLE
#59 Implement backend images

### PR DESCRIPTION
**Pull request by seb8stien1**

### CODE REVIEW

**Review by abhijeetpraveen — APPROVED** [review]
LGTM

**Review by GianlucaP106 — APPROVED** [review]

**Comment by mattcab2002** [comment]
Opened the branch up again because although the first PR works I think other logic is needed.

**Comment by mattcab2002** [comment]
Should have commented before merging, my b

File: mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/image/ImageDataController.java [hunk 1]
```diff
@@ -19,28 +19,56 @@ public class ImageDataController {
...
+    @PostMapping("profile")
+    public ResponseEntity<?> uploadProfileImage(@RequestParam("image") MultipartFile file) throws IOException {
         String name = SecurityContextHolder.getContext().getAuthentication().getName();
-        ImageData image = imageDataService.getInfoByImageByName(name);
+
+        imageDataService.uploadImage(file, "profile_" + name);
+
+        return new ResponseEntity(HttpStatus.OK);
+    }
+
```
Comment (mattcab2002) [inline mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/image/ImageDataController.java:39]: Can we add some put methods for the 2 methods above so we can update the image in the db if it already exists? Currently, we just save it to the db along with the others so when we get them by the username it returns more than one which i believe is causing an error on the backend.

File: mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/image/ImageDataController.java [hunk 2]
```diff
@@ -19,28 +19,56 @@ public class ImageDataController {
...
+        ImageData image = imageDataService.getInfoByImageByName("profile_" + name);
+
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(image);
+    }
+
+    @GetMapping("schedule")
+    public ResponseEntity<?>  getScheduleImageByName(){
+        String name = SecurityContextHolder.getContext().getAuthentication().getName();
+        byte[] image = imageDataService.getImage("schedule_" + name);
```
Comment (mattcab2002) [inline mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/image/ImageDataController.java:61]: diff. status code if the image for the user does not exist in the db? perhaps 204 "No Content". Perhaps this logic should go in the service? Not too familiar with Optional type in Java.

File: mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/image/ImageDataController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                .body(image);
+    }
+
+    @GetMapping()
+    public ResponseEntity<?>  getImageByName(){
+        String name = SecurityContextHolder.getContext().getAuthentication().getName();
+        byte[] image = imageDataService.getImage(name);
+
+        return ResponseEntity.status(HttpStatus.OK)
+                .contentType(MediaType.valueOf("image/png"))
```
Comment (mattcab2002) [inline mcgill-augmented-reality-backend/src/main/java/mcgillar/backend/controllers/image/ImageDataController.java:74]: do they always come out as png's?

**Pull request merged**